app/assets/stylesheets/govuk_publishing_components/components/_layout-super-navigation-header.scss

```diff
@@ -825,29 +825,26 @@ $search-width-or-height: $black-bar-height;
 
 // Dropdown menu items.
 .gem-c-layout-super-navigation-header__dropdown-list-item {
-  margin: 0 0 0 $chevron-indent-spacing;
-  padding: govuk-spacing(2) 0;
+  box-sizing: border-box;
+  padding: 0 0 govuk-spacing(4) 0;
```

Comment:
I don't think that this pattern is more readable and it also adds unnecessary repetition to the CSS. 

If you feel strongly that this should be the way things are done then please suggest a change to the linter.

Reply:
I've got some arguments against this comment:
- I still feel that this is more readable over the shorthand approach because we are communicating more clearly that we are resetting the padding and then adding a padding bottom. I also think this is simpler as a dev doesn't have to do the little calculation in their head that shorthand spacing goes clockwise. I'm not sure what you mean by repetition but efficiency doesn't really matter here since it's all going to get bundled anyway.
- I'm not sure if there's a linting rule for this specifically. I'm aware that you've linked me one for reduced redundancy but is there one for this specific opinion on communicating what we're doing? I haven't been able to fine one.
- Whilst I appreciate you don't agree with my suggestions on separating bits of spacing out, I don't see what's wrong with my suggestion for line 985 (now 998). You've used similar 2 part shorthand elsewhere in this change so I don't see what's wrong with this suggestion.

If you really don't want to do this then I'll not block this, however I still think that the padding under `gem-c-layout-super-navigation-header__popular-item` is still worth changing and that this isn't the same suggestion I made in the previous PR so suggesting a linting change isn't as appropriate a response, in my opinion.

Reply:
Oooh sorry I missed this discussion. Nice work both! Thank you @owenatgov for being so thorough and thinking of readability, I am always in favour of that!

I think this is the relevant linting rule https://stylelint.io/user-guide/rules/list/declaration-block-no-redundant-longhand-properties 

We've not overridden this in https://github.com/alphagov/stylelint-config-gds so wouldn't the linter have caught this? Or is it not that strict/good?

Also my gut reaction was that I thought the shorthand was the accepted industry standard?

In any case, I would always advocate for sticking closer to standards unless we collectively (as a larger group), actively decide to go against any particular rules.

What's the other PR you both mention?

Reply:
@owenatgov I do appreciate your thoroughness in this - whether I disagree with you or not doesn't change that - and this is an interesting solution that you're suggesting. I've thought about this lot[^1] and want to explain my thinking.

At the moment and as far as I can tell[^2], the minifier being used doesn't seem to de-dupe declarations - so this suggestion would result in more CSS being downloaded by the user. Until that problem is solved I wouldn't be happy shipping this suggestion - maybe this is a topic for an issue, which I'm happy to raise.

Secondly, we use BEM[^3] - and one of the benefits is that the order of the classes in CSS can be reordered without any side effects because they all have the same specificity. So the underlying philosophy of BEM is that the order of the CSS shouldn't matter.

Having multiple declarations in a single declaration block means that the order of declarations does matter - so this goes against the grain of how we should be writing CSS. Having order in which the selectors are written not matter, but the order in which the declarations matter seems counter-intuitive.

Thirdly, for me the mix of shorthand and longhand declarations makes it less readable - but I appreciate that this is subjective. As you've said previously, this is [personal preference](https://github.com/alphagov/govuk_publishing_components/pull/2412#pullrequestreview-797827923). So we're back to updating the linting rules.

Updating the linting rules would give everyone confidence that this has been discussed with and validated by the community, as well as making sure that the agreed rules will be adhered to in the future. After all, the main benefit of updating the linter is that it can't be ignored - so the hard work done to make the code more readable doesn't fade over time.

I'd like to reiterate that I'm not saying that this idea shouldn't be pursued - only that this pull request isn't the place to do so. I'm more than happy to support you in any way I can in getting this idea in front of the community so that this idea isn't forgotten and lost.

[^1]: Mainly thanks to a nocturnal teething baby.
[^2]: Turning on production mode locally minified the CSS, but didn't de-dupe it. I wouldn't be surprised if there's something special turned on in production proper though.
[^3]: I should say that proper use of BEM is _strived_ for - but that it's not always possible with the legacy stuff that's around.

Reply:
Thanks for the response Ian. That comment about BEM is especially salient as my opinion here is reliant on the fact that the CSS is in a particular order. It occurs to me that I might be trying to overcomplicate a perfectly valid shorthand standard.

This particular approach (breaking out the shorthand) I'll drop but using 3 part shorthand (eg: `5px 0 10px` instead of `5px 0 10px 0`) I'll run past the community to see what they think.